Directory.Build.targets

```diff
@@ -1,6 +1,14 @@
-<Project>
-    <!--
-        This file is intentionally included even though it is blank.
-        This prevent any Directory.Build.targets file imports above the repo root.
-    -->
+﻿<Project InitialTargets="EnsureKoreBuildRestored">
+  <Target Name="EnsureKoreBuildRestored" Condition=" '$(KoreBuildRestoreTargetsImported)' != 'true' AND '$(MSBuildProjectName)' != 'BenchmarkDotNet.Autogenerated'">
+    <PropertyGroup>
+      <_BootstrapperFile Condition=" $([MSBuild]::IsOSUnixLike()) ">build.sh</_BootstrapperFile>
+      <_BootstrapperFile Condition="! $([MSBuild]::IsOSUnixLike()) ">build.cmd</_BootstrapperFile>
+      <_BootstrapperError>
+        Package references have not been pinned. Run './$(_BootstrapperFile) /t:Pin'.
+        Also, you can run './$(_BootstrapperFile) /t:Restore' which will pin *and* restore packages. '$(_BootstrapperFile)' can be found in '$(MSBuildThisFileDirectory)'.
+      </_BootstrapperError>
+    </PropertyGroup>
+
+    <Error Code="KRB1001" Text="$(_BootstrapperError.Trim())" />
```

Comment:
😭😢😿

For the love of contributors, we must keep the following scenario working.

1. Clone
2. Open solution in Visual Studio
3. Build

Reply:
On a scale from 1 to 10, how important is this to you? When we discussed this in the design meeting with the runtime-eng team, we decided (though regrettable), that this requirement ok. Since we require contributors to ensure `build.cmd` passes anyways, we figured there wasn't much additional burden on contributors or devs to run this warmup command.

Reply:
Hmm, it seems work. How does it know what version of `Internal.AspNetCore.Sdk` to restore?

Reply:
> Hmm, it seems work. 

Even after `git clean -xfd`?

> How does it know what version of Internal.AspNetCore.Sdk to restore?

Same as the other packages: the "lineup" determines this package version. Internal.AspNetCore.Sdk specifically comes from the version bundled in KoreBuild. https://github.com/aspnet/BuildTools/pull/367

Reply:
> Even after git clean -xfd?

Yep. Can't figure out how. 😄

Reply:
It may be an in-memory caching issue with the project system. When I `git clean -xfd` my branch on open a new VS instance, this is what I see:

![image](https://user-images.githubusercontent.com/2696087/29789676-afb70e02-8beb-11e7-8f4f-430f90363296.png)
![image](https://user-images.githubusercontent.com/2696087/29789688-b5c4cd7a-8beb-11e7-8b80-d29324040263.png)

Reply:
Doh, I didn't have the commit in the branch.

Reply:
I'm still very much against this since IMHO it will drive contributors away, but the error is clear, so it's only a single paper cut.

Reply:
By the way, I'm hoping this requirement won't be around forever, but we need a few improvements to VS and NuGet first. They're currently backlogged. https://github.com/NuGet/Home/issues/4781

Reply:
> I'm still very much against this

It's not my favorite either. I'm definitely open to more feedback on this. Any ideas for how to make this action automatic on VS open? I looked for extensibility points, but couldn't find anything reliable. $(SolutionName).sln.{before,after}.targets didn't work because VS has it's own build manager, and there isn't a pre-restore hook that I could find.